state/charm.go

```diff
@@ -28,7 +28,13 @@ import (
 
 // MacaroonCache is a type that wraps State and implements charmstore.MacaroonCache.
 type MacaroonCache struct {
-	*State
+	Charmer
```

Comment:
Once #12414 lands, you'll need to rebase, as it's been done there.

Reply:
Updated.